### ISSUE TITLE
IDE0058 suggestion: option to specify a whitelist of methods to ignore

### ISSUE THREAD

**Issue by Drake53** [dotnet/roslyn#69315]
I use IDE0058 more as a quality rule that a style rule, for example a common pitfall might be the following code:
```csharp
var s = "Hello";
s.Replace("Hello", "World");
```

Since strings are immutable, this code doesn't do anything, it should instead be:
```csharp
var s = "Hello";
s = s.Replace("Hello", "World");
```

Another example I have encountered:
```csharp
DateTime.TryParse(..., out var dateTime);
return dateTime;
```

Rule IDE0058 helps me identify these possible issues, however I cannot use the rule effectively because there are a lot of 'false positives'. For example, `StringBuilder.AppendLine()`, which returns the StringBuilder are thereby triggers this rule. To fix this I have to change my code to:
```csharp
_ = sb.AppendLine(s1);
_ = sb.AppendLine(s2);
...
```

Having to use the discard on every line reduces the readability. I'd prefer to maintain a list of methods for which it's known that the return value can be safely (implicitly) discarded, without using `_`. This could be achieved in a similar way as the BannedSymbols analyzer, where you have (a) separate file(s) containing a list of methods:
```
System.Text.StringBuilder.AppendLine(System.String)
// etc
```
